### PR TITLE
fix/routing: treat bootstrapping to any node the same

### PR DESCRIPTION
**Pull request by jeanphilippeD**
Addressing comment https://github.com/maidsafe/routing/pull/1954#discussion_r361638168 from PR merged as https://github.com/maidsafe/routing/pull/1983.

Remove first node special case if not enough elders.

Test:
Update unit test
soak test + clippy

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -69,5 +69,4 @@ pub enum RoutingError {
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
 pub enum BootstrapResponseError {
```
Comment (maqi) [inline src/error.rs]: probably no need to make this as `enum` anymore ? 
as you will only be `NotApproved`

File: src/states/elder/tests.rs [hunk 2]
```diff
@@ -503,23 +503,25 @@ fn when_accumulate_offline_and_start_dkg_and_section_info_then_node_is_removed_f
...
 #[test]
-fn accept_previously_rejected_node_after_reaching_elder_size() {
+fn accept_node_before_and_after_reaching_elder_size() {
     // Set section size to one less than the desired number of the elders in a section. This makes
     // us reject any bootstrapping nodes.
     let mut elder_test = ElderUnderTest::new(ELDER_SIZE - 1);
     let node = JoiningNodeInfo::with_addr(&mut elder_test.rng, "198.51.100.0:5000");
 
-    // Bootstrap fails for insufficient section size.
+    // Bootstrap succeed even with to few elders.
```
Comment (maqi) [inline src/states/elder/tests.rs]: `too` few

File: src/states/elder/tests.rs [hunk 3]
```diff
@@ -503,23 +503,25 @@ fn when_accumulate_offline_and_start_dkg_and_section_info_then_node_is_removed_f
...
     elder_test.handle_bootstrap_request(*node.public_id(), node.connection_info());
-    assert!(!elder_test.is_connected(&node.connection_info().peer_addr));
+    assert!(elder_test.is_connected(&node.connection_info().peer_addr));
+
+    let node = JoiningNodeInfo::with_addr(&mut elder_test.rng, "198.51.100.1:5000");
```
Comment (maqi) [inline src/states/elder/tests.rs]: this is actually a new node.
which means the node at address 198.51.100.0 could be dropped? (or at least indicates that as using the same variable name.)
maybe better use a different name?

**Comment by jeanphilippeD** [comment]
Fixed review comments.

**Pull request merged**